core/src/VolatilityOracle.sol

```diff
@@ -0,0 +1,181 @@
+
+                uint256 maxChange = timeSinceLastWrite * IV_CHANGE_PER_SECOND;
+                if (iv > lastWrite.iv + maxChange) iv = lastWrite.iv + maxChange;
+                else if (iv + maxChange < lastWrite.iv) iv = lastWrite.iv - maxChange;
+            }
+
+            // Store the new feeGrowthGlobals sample and update `lastWrites`
+            uint8 next = uint8((lastWrite.index + 1) % 60);
+            arr[next] = b;
+            lastWrites[pool] = LastWrite(next, uint32(block.timestamp), uint216(iv));
+
+            return (data.sqrtMeanPriceX96, iv);
+        }
+    }
+
+    function consult(IUniswapV3Pool pool) external view returns (uint160, uint256) {
+        (uint160 sqrtMeanPriceX96, ) = Oracle.consult(pool, ORACLE_LOOKBACK);
+        return (sqrtMeanPriceX96, lastWrites[pool].iv);
+    }
+
+    function _getPoolMetadata(IUniswapV3Pool pool) private view returns (Volatility.PoolMetadata memory metadata) {
+        (, , uint16 observationIndex, uint16 observationCardinality, , uint8 feeProtocol, ) = pool.slot0();
+        // If block times are inconsistent, `maxSecondsAgo` from oracle may be inflated. Divide by 2 to be extra safe.
+        metadata.maxSecondsAgo = Oracle.getMaxSecondsAgo(pool, observationIndex, observationCardinality) / 2;
+        require(metadata.maxSecondsAgo > ORACLE_LOOKBACK, "Aloe: cardinality");
+
+        uint24 fee = pool.fee();
+        metadata.gamma0 = fee;
+        metadata.gamma1 = fee;
+        unchecked {
+            if (feeProtocol % 16 != 0) metadata.gamma0 -= fee / (feeProtocol % 16);
+            if (feeProtocol >> 4 != 0) metadata.gamma1 -= fee / (feeProtocol >> 4);
+        }
+
+        metadata.tickSpacing = pool.tickSpacing();
+    }
+
+    function _getPoolData(IUniswapV3Pool pool) private view returns (Volatility.PoolData memory data) {
+        (uint160 sqrtPriceX96, int24 currentTick, , , , , ) = pool.slot0();
+        (uint160 sqrtMeanPriceX96, uint160 secondsPerLiquidityX128) = Oracle.consult(pool, ORACLE_LOOKBACK);
+        data = Volatility.PoolData(
+            sqrtPriceX96,
+            currentTick,
+            sqrtMeanPriceX96,
+            secondsPerLiquidityX128,
+            ORACLE_LOOKBACK,
+            pool.liquidity()
+        );
+    }
+
+    function _getFeeGrowthGlobalsNow(IUniswapV3Pool pool) private view returns (Volatility.FeeGrowthGlobals memory) {
+        return
+            Volatility.FeeGrowthGlobals(
+                pool.feeGrowthGlobal0X128(),
+                pool.feeGrowthGlobal1X128(),
+                uint32(block.timestamp)
+            );
+    }
+
+    function _getFeeGrowthGlobalsOld(
+        Volatility.FeeGrowthGlobals[60] storage arr,
+        uint256 index
+    ) private view returns (Volatility.FeeGrowthGlobals memory) {
+        uint256 target = block.timestamp - 1 hours;
+
+        // See if the newest sample is best
+        Volatility.FeeGrowthGlobals memory sample = arr[index];
+        if (sample.timestamp <= target) return sample;
```

Comment:
`Volatility.FeeGrowthGlobals[60] storage arr` will have multiple values. If the timestamp of the newest sample is less than the `target` time, we return the `sample` because it's best. "Best" in this means "most recent".

Reply:
"Best" is intended to mean "nearest to `target`". I'll update the comments.